frontend/src/app/pages/signup/signup.page.component.ts

```diff
@@ -12,24 +21,52 @@ import { SuccessfulUserAuth } from '../../core/models/auth.interface';
 @Component({
     selector: 'signup-page',
     standalone: true,
-    imports: [CommonModule],
+    imports: [CommonModule, ReactiveFormsModule, CardModule, ButtonModule, FormsModule, PasswordModule, CheckboxModule],
     templateUrl: './signup.page.component.html',
     styleUrl: './signup.page.component.css'
 })
 export class SignUpPageComponent {
-    title = 'apluschiever'
+    title = 'Sign-Up Page'
 
-    constructor(private oauthService: OAuth2Service) { }
+    registerForm = this.fb.group({
+        email_address: ['', [Validators.required, Validators.email]],
+        first_name: ['', [Validators.required, Validators.pattern(/^[a-zA-Z]+(?: [a-zA-Z]+)*$/)]],
+        last_name: ['', [Validators.required, Validators.pattern(/^[a-zA-Z]+(?: [a-zA-Z]+)*$/)]],
+        password: ['', Validators.required],
+        confirmPassword: ['', Validators.required]
+      }, {
+        validators: passwordMatchValidator
+      })
 
+    constructor(private router: Router, private fb: FormBuilder, private oauthService: OAuth2Service) { }
+    get first_name() {
+        return this.registerForm.controls['first_name'];
+      }
+      get last_name() {
+        return this.registerForm.controls['last_name'];
+      }
+    
+      get email_address() {
+        return this.registerForm.controls['email_address'];
+      }
+    
+      get password() {
+        return this.registerForm.controls['password'];
+      }
+    
+      get confirmPassword() {
+        return this.registerForm.controls['confirmPassword'];
+      }
     public signup(): void {
         // Sample of how to sign up
-        this.oauthService.sign_up({
-            email_address: 'hello-world@gmail.com',
-            password: 'SAMPLE',
-            first_name: 'TEST FIRST NAME',
-            last_name: 'TEST LAST NAME'
-        }).subscribe((res: SuccessfulUserAuth) => {
+        const postData = { ...this.registerForm.value };
+        this.oauthService.sign_up(postData as SignUpInfo).subscribe((res: SuccessfulUserAuth) => {
             console.log(res);
+            this.oauthService.sign_in( postData as SignInInfo ).subscribe((res: SuccessfulUserAuth) => {
```

Comment:
In the future we will need to add error handling to signup and login pages, but for now this is good. 

Error handling on these things can be a milestone 2/3 thing.